src/wallet/rpcwallet.cpp

```diff
@@ -2806,8 +2806,8 @@ static RPCHelpMan unloadwallet()
 {
     std::string wallet_name;
     if (GetWalletNameFromJSONRPCRequest(request, wallet_name)) {
-        if (!request.params[0].isNull()) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot unload the requested wallet");
+        if (!(request.params[0].isNull() || request.params[0].get_str() == wallet_name)) {
```

Comment:
2e9ada51830236a027371a48697b96 nit, would find this clearer
```suggestion
        if (!request.params[0].isNull() && request.params[0].get_str() != wallet_name) {
```

Reply:
That's strictly less clear (it relies on implicit precedence order)